### PR TITLE
fixes SPF and DMARC prefix checks (#252 #253)

### LINKED ISSUE #252: Prefix in DMARC module differs from RFC
Hello,

while using the DMARC module, we found that its implementation [modules/dmarc/dmarc.go#L56](https://github.com/zmap/zdns/blob/ed5b6828b56753f42903a3b7b82ade161e65ec02/modules/dmarc/dmarc.go#L56) checks only the first 7 characters ("v=DMARC") of DMARC TXT record and not the ABNF defined in [RFC 7489](https://datatracker.ietf.org/doc/html/rfc7489) as "v=DMARC1" with optional whitespaces.

This can lead to cases when a DMARC of a domain is not set up correctly (e.g. "v=DMARCv1") and zdns' result status is "NOERROR" even though the DMARC configuration is not valid.

Is it an intentional behviour?

Thanks.

### PR DESCRIPTION
**Pull request by bayerhonza**
resolves #252 #253 

Added prefix checks for TXT records (DMARC and SPF) based on regular expressions

DMARC prefix is now checked correctly as well as case-insensitive SPF prefix

### CODE REVIEW

File: modules/dmarc/dmarc.go [hunk 1]
```diff
@@ -18,8 +18,11 @@ import (
 	"github.com/miekg/dns"
 	"github.com/zmap/zdns"
 	"github.com/zmap/zdns/modules/miekg"
+	"regexp"
 )
 
+const dmarcPrefixRegexp = "v[\x09\x20]*=[\x09\x20]*DMARC1[\x09\x20]*;[\x09\x20]*"
```
Comment (dadrian) [inline modules/dmarc/dmarc.go]: Are \x09 and \x20 whitespace characters? Should we instead be using `[\s]*`?

**Review by dadrian — COMMENTED** [review]
Sorry for the delay on the review.

**Comment by spencerdrak** [comment]
This also looks fine to me pending the resolution of @dadrian's comment above, assuming the only acceptable whitespace that might be encountered is tab (`\x09`) and space (`\x20`). `\s` will also catch newlines as whitespace, so it may be more useful to only catch tabs and spaces.

File: modules/dmarc/dmarc.go (hunk 1 above)
Comment (bayerhonza) [reply modules/dmarc/dmarc.go]: I used the ABNF from the [DMARC RFC](https://datatracker.ietf.org/doc/html/rfc7489) that states:

`dmarc-version   = "v" *WSP "=" *WSP %x44 %x4d %x41 %x52 %x43 %x31`

WSP comes from [ABNF RFC](https://datatracker.ietf.org/doc/html/rfc5234) and is defined as:

` WSP            =  SP / HTAB
                                ; white space`

`[\s]*` is not equal to `[\x09\x20]*` as the first one accepts also newline, form feed, and carriage return. [[1]](https://pkg.go.dev/regexp/syntax)

However, I now see that `"v"` should actually be case insensitive as ABNF strings are case insensitive [[ABNF 2.3]](https://datatracker.ietf.org/doc/html/rfc5234). So the code should rather look like this:

`const dmarcPrefixRegexp = "[vV][\x09\x20]*=[\x09\x20]*DMARC1[\x09\x20]*;[\x09\x20]*"`

What do you think?

Comment (spencerdrak) [reply modules/dmarc/dmarc.go]: I agree with that - @dadrian you have any thoughts?

Comment (zakird) [reply modules/dmarc/dmarc.go]: Hard to argue against matching the RFC! Let's update to case insensitive and ship it!

**Comment by bayerhonza** [comment]
I updated the code as discussed: "V" in DMARC prefix is now case-insensitive.

**Comment by spencerdrak** [comment]
@bayerhonza we merged in some other work and looks like we have some conflicts now. If you don't mind taking a look at that, we can merge once the conflicts are resolved. Otherwise, I'll pull a copy and sort out the conflicts when I've got the chance. 

Thanks! Sorry for the delay.

**Comment by bayerhonza** [comment]
I resolved the conflicts. I think we can either close this PR or #265 as there are duplicate. I only created #265 because I added a testing library.

**Review by dadrian — APPROVED** [review]

**Pull request merged**